<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="use_rviz" default="False" />
    <arg name="use_camera" default="False" />
    <arg name="image_width" default="1920" />
    <arg name="image_height" default="1080" />
    <arg name="camera" default="/camera/color/image_raw" />
    <arg name="camera_info" default="/camera/color/camera_info" />
    <arg name="tf_prefix" default="marker_id" />
    <arg name="show_detections" default="true" />
    <arg name="marker_size" default="0.5" doc="length of the physical marker's side in meters"/>
    <!-- <arg name="dictionary_name" default="DICT_5X5_50" /> -->
    <arg name="dictionary_name" default="DICT_6X6_250" />
    <arg name="blur_window_size" default="7" />

   <!--   usb_cam node -->
  <group if="$(arg use_camera)" >
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
      <param name="video_device" value="/dev/video1" />
      <param name="image_width" value="$(arg image_width)" />
      <param name="image_height" value="$(arg image_height)" />
      <param name="pixel_format" value="yuyv" />
      <param name="camera_frame_id" value="usb_cam" />
      <param name="io_method" value="mmap"/>
    </node>
  </group>

  <!-- 1920x1080  -->
  <node if="$(eval arg('image_width') ==1920)" name="camera_info" pkg="rostopic" type="rostopic"
    args="pub /camera/camera_info sensor_msgs/CameraInfo
   '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera1'},
    height: 1080, width: 1920, distortion_model: 'plumb_bob',
    D: [0.061957, -0.124832, 0.002573, -0.004753, 0.0],
    K: [1179.598752, 0.0, 928.099247, 0.0, 1177.000389, 558.635461, 0.0, 0.0, 1.0],
    R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
    P: [1156.897217, 0.0, 908.598013, 0.0, 0.0, 1188.634277, 561.054263, 0.0, 0.0, 0.0, 1.0, 0.0],
    binning_x: 0, binning_y: 0,
    roi: {x_offset: 0, y_offset: 0, height: 1080, width: 1920, do_rectify: false}}' -r 2"/>
      <node pkg="aruco_detector" type="node" name="aruco_detect_node" output="screen">
          <param name="camera" type="string" value="$(arg camera)"/>
          <param name="camera_info" type="string" value="$(arg camera_info)"/>
          <param name="tf_prefix" type="string" value="$(arg tf_prefix)"/>
          <param name="show_detections" type="bool" value="$(arg show_detections)"/>
          <param name="marker_size" type="double" value="$(arg marker_size)"/>
          <param name="dictionary_name" type="string" value="$(arg dictionary_name)"/>
          <param name="image_width" type="double" value="$(arg image_width)"/>
          <param name="image_height" type="double" value="$(arg image_height)"/>
          <param name="blur_window_size" type="double" value="$(arg blur_window_size)"/>
  </node>

<!-- RVIZ -->
  <group if="$(arg use_rviz)" >
    <node pkg="rviz" type="rviz" name="markers_extrinsic" args="-d $(find aruco_detector)/config/aruco_opencv.rviz"/>
  </group>
</launch>
